### ISSUE TITLE
scikit-image 0.17.1 breaks image segmentation

### ISSUE THREAD

**Issue by jklaise** [SeldonIO/alibi#215]
Anchor image test currently doesn't pass on the latest `skimage` version (https://github.com/scikit-image/scikit-image/issues/4675).

**Comment by rfezzani** [comment]
I checked and it seems that `skimage` v0.16 is also affected:
```python
>>> from skimage.segmentation import slic
>>> import numpy as np
>>> im = np.random.rand(28, 28).astype(np.float32)
>>> slic(im)
...
  File "skimage/segmentation/_slic.pyx", line 13, in skimage.segmentation._slic._slic_cython
ValueError: Buffer dtype mismatch, expected 'double' but got 'float'
>>> skimage.__version__
'0.16.2'
```
Have you introduced any other change apart `skimage` version bump?

**Comment by rfezzani** [comment]
I opened scikit-image/scikit-image#4683 to fix this issue...
Is there any numpy version change in your tests BTW?

**Comment by jklaise** [comment]
@rfezzani thanks, I've narrowed down the issue to having `pooch` as a dependency for `scikit-image=0.17.1`.If I install `scikit-image=0.16.2` without `pooch` then the tests pass.

**Comment by jklaise** [comment]
0.17.2 of scikit-image fixes this issue so we just say that 0.17.1 is not allowed in #283 .
